### DISCUSSION TITLE
Configuration of certificate/tls with shovel and federation in case of Rabbitmq CRD

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#8970] [Questions] [answered]
I have deployed  HA Rabbitmq cluster using operators (RMQ cluster kubernetes operator and messaging topology operator). We have provided the tls certificate for webhooks in messaging topology operator.

I have 2 rabbitmq setup, one for internal and one for external between which I would like to apply federation and shoveling.

But I am getting how to configure certificate/tls with shovel and federation in above setup?

**Comment by michaelklishin** [comment, marked as answer]
Search for "TLS" in

 * [This Federation guide section](https://rabbitmq.com/federation.html#tls-connections)
 * [The guide on dynamic Shovels](https://rabbitmq.com/shovel-dynamic.html#declaring)
 
TLS-related properties are provided as part of the Shovel URI and upstream definition for Federation.

**Reply by paywagh11** [reply]
In the URI we have to give path to the cert files and key file so just want to confirm we have to mount these certificate in the RMQ cluster right ?? amqps://myhost?cacertfile=/path/to/ca_certificate.pem   &certfile=/path/to/client_certificate.pem   &keyfile=/path/to/client_key.pem   &verify=verify_peer   &server_name_indication=myhost

**Reply by mkuratczyk** [reply]
Yes, you need the files to exist.
